spec/brainstem/presenter_spec.rb

```diff
@@ -680,37 +680,80 @@ def preloader_should_receive(hsh)
     end
   end
 
-  describe "#apply_ordering_to_scope" do
+  describe '#apply_ordering_to_scope' do
     let(:presenter_class) { WorkspacePresenter }
     let(:presenter) { presenter_class.new }
     let(:scope) { Workspace.where(nil) }
 
     it 'uses #calculate_sort_name_and_direction to extract a sort name and direction from user params' do
-      presenter_class.sort_order :title, "workspaces.title"
+      presenter_class.sort_order :title, 'workspaces.title'
       mock(presenter).calculate_sort_name_and_direction('order' => 'title:desc') { ['title', 'desc'] }
       presenter.apply_ordering_to_scope(scope, 'order' => 'title:desc')
     end
 
-    it 'runs procs in the context of any helpers' do
-      presenter_class.helper do
-        def some_method
+    context 'when the sort is a proc' do
+      it 'runs procs in the context of any helpers' do
+        presenter_class.helper do
+          def some_method
+          end
+        end
+
+        direction = nil
+        presenter_class.sort_order(:title) do |scope, d|
+          some_method
+          direction = d
+          scope
+        end
+
+        presenter.apply_ordering_to_scope(scope, 'order' => 'title:asc')
+        expect(direction).to eq 'asc'
+      end
+
+      it 'can chain multiple sorts together' do
+        presenter_class.sort_order(:title) do |scope|
+          scope.order('workspaces.title desc').order('workspaces.id desc')
+        end
+
+        sql = presenter.apply_ordering_to_scope(scope, 'order' => 'title').to_sql
+        expect(sql).to match(/order by workspaces.title desc, workspaces.id desc/i)
```

Comment:
We tried; it got kind of crazy examining the arel meta-objects that get created when you pass in symbols.  Also if you pass in a sort of just `id` without the table name, and some other weirdo edge cases.